controllers/members.js

```diff
@@ -99,9 +101,47 @@ const archiveMembers = async (req, res) => {
   }
 };
 
+const updateRoles = async (req, res) => {
+  try {
+    const userId = req.params.id;
+    const user = await fetchUser({ userId });
+    if (user?.userExists) {
+      const dataToUpdate = req.body;
+      const successObject = await members.updateRoles(user.user.id, dataToUpdate);
```

Comment:
I'm thinking why are we passing only userId and fetching the user again, can we pass the user object, so that we don't fetch it again.
Let others comment @RitikJaiswal75 @vinayak-trivedi

Reply:
okk I will ping once @vinayak-trivedi for this on discord

Reply:
don't make it this generic please, add better names for this

Reply:
> don't make it this generic please, add better names for this

You are talking about change name `successObject` to better variable name

Reply:
> regarding your suggestion @vikhyat187, it will definitely save us few reads in the database, I think that this will not be a good to do, so let's take a hypothetical example, where in future we have two super users and we both of them make a requests to make the user a member at the same time, so ideally if the user is already a member we should return a response that user is already a member from backend, which will be possible only if we fetch the user details, I might not be clear with my description, please let me know if you want to discuss this

I think we can add optimistic locking, to only allow one update from a set of parallel updates, @iamitprakash please share your point of view to tackle this.

@vinayak-trivedi Let's connect today around 9:30PM IST

Reply:
Done